plonkish_backend/src/accumulation/protostar/ivc/aggregation/agg_circuit.rs

```diff
@@ -154,153 +155,224 @@ impl CircuitExt<grumpkin::Fr> for SecondaryAggregationCircuit {
-//         //     secondary_output.into_iter().flatten(),
-//         //     pairing_acc.into_iter().flatten(),
-//         // ]
-//         // .enumerate()
-//         // {
-//         //     layouter.constrain_instance(cell_map[&witness.id()].cell(), chip.instance, idx)?;
-//         // }
-
-//         Ok(())
-//     }
-// }
-
-// impl CircuitExt<bn256::Fr> for PrimaryAggregationCircuit {
-//     fn instances(&self) -> Vec<Vec<bn256::Fr>> {
-//         vec![self.instances.clone()]
-//     }
-// }
\ No newline at end of file
+    //todo fix this with other synthesizes
+    fn synthesize(
+        &self,
+        config: Self::Config,
+        mut layouter: impl Layouter<bn256::Fr>,
+    ) -> Result<(), Error> {
+
+        let mut builder = RangeCircuitBuilder::from_stage(CircuitBuilderStage::Keygen)
+        .use_k(8)
+        .use_lookup_bits(9);
+
+        let range = builder.range_chip();
+        let gate_chip = GateChip::<bn256::Fr>::new();
+        let base_chip = FpChip::<bn256::Fr, bn256::Fq>::new(&range, NUM_LIMB_BITS, NUM_LIMBS);
+        let native_chip = NativeFieldChip::new(&range);
+        // Error Correcting Chip 
+        let ecc_chip = EccChip::new(&native_chip);
+
+        let mut pool = mem::take(builder.pool(0));
+        
+        // Chip here is built with a combination of chips ß
+        /*
+        
+            Initially passing three arguments, but requires only a range chip. 
+            let range passed
+         */
+        let chip = strawman::Chip::<bn256::G1Affine>::create(range);
+
+        // let chip =
+        //     <strawman::Chip<bn256::G1Affine> as TwoChainCurveInstruction<bn256::G1Affine>>::new(chip,
+        //         config,
+        //     );
+
+        let mut builder = RangeCircuitBuilder::from_stage(CircuitBuilderStage::Keygen)
+            .use_k(8)
+            .use_lookup_bits(9);
+
+
+        // Why instantiate again?
+        // let mut pool = mem::take(builder.pool(0));
+
+
+        let aggregator = ProtostarIvcAggregator::new(
+            self.vp_digest,
+            self.vp.clone(),
+            self.primary_arity,
+            chip.clone(),
+            chip.clone(),
+        );
+
+        let mut transcript = strawman::PoseidonTranscriptChip::new(
+            builder.main(0),
+            strawman::decider_transcript_param(),
+            chip.clone(),
+            self.proof.clone(),
+        );
+
+        let (primary_num_steps, primary_initial_input, primary_output, h_ohs_from_last_nark) =
+            aggregator.verify_ipa_grumpkin_ivc_with_last_nark(
+                &mut pool,
+                self.num_steps,
+                self.initial_input.clone(),
+                self.output.clone(),
+                self.acc_before_last.clone(),
+                self.last_instance,
+                &mut transcript,
+            )?;
+
+
+        /*
+           Question:
+            1. Can we use the same circuit aggregator and transcript for the secondary aggregation?
```

Comment:
aggregator and transcript are chips which take a scalarfield. for primary and seconadry agg circuit, we instance them with different scalarfields, i.e bn254 and grumpkin respectively.